src/main/bash/sdkman-path-helpers.sh

```diff
@@ -50,28 +50,38 @@ function __sdkman_set_candidate_home() {
 }
 
 function __sdkman_export_candidate_home() {
-	local candidate_name="$1"
-	local candidate_dir="$2"
-	local candidate_home_var="$(echo ${candidate_name} | tr '[:lower:]' '[:upper:]')_HOME"
-	export $(echo "$candidate_home_var")="$candidate_dir"
-}
+	local candidate_name="$1" candidate_dir="$2" upcase_name
 
-function __sdkman_determine_candidate_bin_dir() {
-	local candidate_dir="$1"
-	if [[ -d "${candidate_dir}/bin" ]]; then
-		echo "${candidate_dir}/bin"
+	# Generate home variable name using shell-specific methods
+	# to avoid heavy-weight fork/exec system calls.
+
+	if [ -n "${ZSH_VERSION-}" ]; then
+		# zsh: uppercase via ${value:u}
+		upcase_name="${candidate_name:u}"
+	elif [ -n "${BASH_VERSION-}" ] && [ "${BASH_VERSINFO[0]:-0}" -ge 4 ] 2>/dev/null; then
```

Comment:
You can also use the [global definition](https://github.com/sdkman/sdkman-cli/blob/master/src/main/bash/sdkman-init.sh#L67) here. Do we need to bolster our global definition with `BASH_VERSIONINFO` too?